### DISCUSSION TITLE
Disabling Observable instrument for some time

### DISCUSSION THREAD

**Discussion by mihi-atti** [open-telemetry/opentelemetry-specification#2678] [Q&A] [answered]
How can the situation be handled when the given sensor is not working but the ObservableGauge instrument callback SHOULD return something? In this case I don't want to push these dummy data to OTEL collector and create timeseries data.
Is there any way to disable the given instrument for some time and enable it again later?

**Comment by jack-berg** [comment, marked as answer]
Observable instruments manage their own time series. Put another way, if an observable instrument stops observing values for a particular series, that series should not show up in the metrics exported by the SDK. In your case, this means that when the sensor is not working, the instrument simply shouldn't observe any measurements.

**Reply by mihi-atti** [reply]
Thanks for your quick answer. Once an ObservableGauge is created there is no way to unregister it from the system in case the sensor is not working later. The callback will be invoked and it SHOULD reply something. I am using the .NET SDK. I put the metrics into the OTEL collector with using the PeriodicExportingMetricReader and OtlpMetricExporter.

**Comment by mihi-atti** [comment]
@jack-berg I got the answer here: https://github.com/open-telemetry/opentelemetry-dotnet/issues/3460
Thanks for your answer. Have a nice day!
